src/radix.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""File implements a radix sort method."""
+from __future__ import unicode_literals
+from collections import OrderedDict
+from collections import deque
+import random
+import timeit
+
+
+def radix_sort(sort_list):
+    """Sorts input based on the least significant digits of the data first
+    and cascades upward until the input has been sorted by it's most
+    signifiant digit.
+
+    Worked with Steven and Victor writing this function."""
+    if not isinstance(sort_list, list):
+        raise TypeError("Radix sort only accepts inputs inside a Python list.")
+    for item in sort_list:
```

Comment:
This loop gives your algorithm an extra O(n) of time complexity. Is it really worth it to type check your input?

You could add the type check in one of your later loops, since you're already looping through the whole input list.